Source/WebCore/platform/mediastream/RealtimeMediaSourceCapabilities.h

```
@@ -114,17 +114,30 @@ class CapabilityValueOrRange {
 template<class Encoder>
 void CapabilityValueOrRange::encode(Encoder& encoder) const
 {
-    encoder.encodeFixedLengthData(reinterpret_cast<const uint8_t*>(&m_minOrValue), sizeof(ValueUnion), alignof(ValueUnion));
-    encoder.encodeFixedLengthData(reinterpret_cast<const uint8_t*>(&m_max), sizeof(ValueUnion), alignof(ValueUnion));
+    encoder.encodeObject(m_minOrValue);
+    encoder.encodeObject(m_max);
     encoder << m_type;
 }
 
 template<class Decoder>
 bool CapabilityValueOrRange::decode(Decoder& decoder, CapabilityValueOrRange& valueOrRange)
 {
-    return decoder.decodeFixedLengthData(reinterpret_cast<uint8_t*>(&valueOrRange.m_minOrValue), sizeof(ValueUnion), alignof(ValueUnion))
-        && decoder.decodeFixedLengthData(reinterpret_cast<uint8_t*>(&valueOrRange.m_max), sizeof(ValueUnion), alignof(ValueUnion))
-        && decoder.decode(valueOrRange.m_type);
+    auto minOrValue = decoder.template decodeObject<ValueUnion>();
+    if (!minOrValue)
+        return false;
+
+    auto max = decoder.template decodeObject<ValueUnion>();
+    if (!max)
+        return false;
+
+    auto type = decoder.template decode<Type>();
+    if (!type)
+        return false;
+
```

The pattern for the decode should be:
```
auto minOrValue = decoder.template decodeObject<ValueUnion>();
auto max = decoder.template decodeObject<ValueUnion>();
auto type = decoder.template decode<Type>();
if (UNLIKELY(!decoder.isValid())
    return false;
```

E.g. the common case be "branch free" from the caller statement list.

---

With respect to the comment above about matching ArgumentCoders to implementations, it'd be good to regularise this eventually to
```
auto minOrValue = decoder.template decode<ValueUnion>();
auto max = decoder.template decode<ValueUnion>();
auto type = decoder.template decode<Type>();
if (UNLIKELY(!decoder.isValid())
    return false;
```

With respect to comment below about legacy decoder, this markup would be
```
(void) decoder.decode(&m_minOrValue);
(void) decoder.decode(&m_max);
(void  decoder.decode(&m_type);
if (UNLIKELY(!decoder.isValid())
    return false;
return true;
```
However I don't think this is that important at this point.

---

I think applying batched, branch-free decoding will be simplest (or already is simplest) in the generated serializers code.

Decoding from an already-invalid decoder perhaps deserves some investigation. The buffer position would be null, and I believe any decoding would fail early in the buffer containment check, but I didn't review thoroughly.